framework/wazuh/core/cluster/common.py

```diff
@@ -1047,6 +1055,334 @@ def get_node(self):
+            a JSON file named files_metadata.json.
+
+        Returns
+        -------
+        bool
+            True if files were correctly sent to the master node, None otherwise.
+        """
+        # Start the synchronization process with the master and get a taskID.
+        task_id = await self.manager.send_request(command=self.cmd, data=b'')
+        if isinstance(task_id, Exception):
+            raise task_id
+        elif task_id.startswith(b'Error'):
+            self.logger.error(task_id.decode())
+            exc_info = json.dumps(exception.WazuhClusterError(3016, extra_message=str(task_id)),
+                                  cls=WazuhJSONEncoder).encode()
+            await self.manager.send_request(command=self.cmd + b'_r', data=b'None ' + exc_info)
+            return
+
+        self.logger.debug(
+            f"Compressing {'files and ' if files_to_sync else ''}'files_metadata.json' of {len(files_metadata)} files."
+        )
+        compressed_data_path = cluster.compress_files(name=self.manager.name, list_path=files_to_sync,
+                                                      cluster_control_json=files_metadata)
+
+        try:
+            # Send zip file to the master into chunks.
+            self.logger.debug("Sending zip file to master.")
+            await self.manager.send_file(filename=compressed_data_path)
+            self.logger.debug("Zip file sent to master.")
+
+            # Finish the synchronization process and notify where the file corresponding to the taskID is located.
+            result = await self.manager.send_request(command=self.cmd + b'_e',
+                                                     data=task_id + b' ' + os.path.relpath(compressed_data_path,
+                                                                                           common.wazuh_path).encode())
+            if isinstance(result, Exception):
+                raise result
+            elif result.startswith(b'Error'):
+                raise exception.WazuhClusterError(3016, extra_message=result.decode())
+            return True
+        except exception.WazuhException as e:
+            # Notify error to master and delete its received file.
+            self.logger.error(f"Error sending zip file: {e}")
+            await self.manager.send_request(command=self.cmd + b'_r',
+                                            data=task_id + b' ' + json.dumps(e, cls=WazuhJSONEncoder).encode())
+        except Exception as e:
+            # Notify error to master and delete its received file.
+            self.logger.error(f"Error sending zip file: {e}")
+            exc_info = json.dumps(exception.WazuhClusterError(1000, extra_message=str(e)),
+                                  cls=WazuhJSONEncoder).encode()
+            await self.manager.send_request(command=self.cmd + b'_r', data=task_id + b' ' + exc_info)
+        finally:
+            os.unlink(compressed_data_path)
```

Comment:
For consistency, since the SyncWazuhdb task is used by both types of nodes, I decided that this task although currently only used by the worker node should be in the common file, this way everything is gathered in the same file and it is also prepared in case in the future it is needed in the master as well.